gui/guihtml/src/TGHtmlUri.cxx

```diff
@@ -350,11 +350,11 @@ char *TGHtml::ResolveUri(const char *zUri)
             zBuf[i] = 0;
          }
          // coverity[secure_coding]
-         strcat(zBuf, term->fZPath);
+         strlcat(zBuf, term->fZPath, strlen(term->fZPath)+1);
          for (i = 0; zBuf[i]; i++) {
             if (zBuf[i] == '/' && zBuf[i+1] == '.' && zBuf[i+2] == '/') {
                // coverity[secure_coding]
-               strcpy(&zBuf[i+1], &zBuf[i+3]);
+               strlcpy(&zBuf[i+1], &zBuf[i+3], strlen(&zBuf[i+3])+1);
```

Comment:
Here totally wrong. I am not sure that `strcpy` or `strlcpy` can correctly handle overlapping memory regions.
The only secure way - copy symbol by symbol in for loop

Reply:
Even `coverity` was detecting it long time ago